### PR TITLE
add shield library

### PR DESCRIPTION
**Pull request by lkotsonis**
Specify library name and version:  **shield/0.1**

I'm acting as a representative of Holoplot to create this PR for `shield`. It is a library we would like to use as a dependency for other open-source repositories that would be coming up.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/shield/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import os
+
+class ShieldConan(ConanFile):
+    name = "shield"
+    version = "0.1"
+    topics = ("shield", "utility", "warnings", "suppression")
+    description = "C++ warning suppression headers."
+    settings = "os", "compiler", "build_type", "arch"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = url
```
Comment (uilianries) [inline recipes/shield/all/conanfile.py:11]:
```suggestion
    homepage = "https://github.com/holoplot/shield/"
```

File: recipes/shield/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
+from conans import ConanFile, tools, CMake
+import os
+
+class ShieldConan(ConanFile):
+    name = "shield"
+    version = "0.1"
```
Comment (uilianries) [inline recipes/shield/all/conanfile.py:6]:
```suggestion
```

do not add version, so you can re-use the same recipe to any version

File: recipes/shield/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    topics = ("shield", "utility", "warnings", "suppression")
```
Comment (uilianries) [inline recipes/shield/all/conanfile.py:7]:
```suggestion
    topics = ("utility", "warnings", "suppression")
```
Do not include project's name

File: recipes/shield/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    description = "C++ warning suppression headers."
+    settings = "os", "compiler", "build_type", "arch"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = url
+    license = "MIT"
+    no_copy_source = True
+    required_conan_version = ">=1.33.0"
```
Comment (uilianries) [inline recipes/shield/all/conanfile.py:14]:
```suggestion
```

It's not an attribute, it should add as a global variable

File: recipes/shield/all/test_package/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 3.1)
+project(test_package)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+add_executable(${PROJECT_NAME} main.cpp)
+target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
```
Comment (uilianries) [inline recipes/shield/all/test_package/CMakeLists.txt:8]:
```suggestion
conan_basic_setup(TARGETS)

find_package(shield CONFIG REQUIRED)

add_executable(${PROJECT_NAME} main.cpp)
target_link_libraries(${PROJECT_NAME} shield::shield)
```

Prefer targets, CONAN_LIBS will be deprecated on Conan 2.0

File: recipes/shield/all/test_package/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+class TestPackageConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
```
Comment (uilianries) [inline recipes/shield/all/test_package/conanfile.py:7]:
```suggestion
    generators = "cmake", "cmake_find_package_multi"
```
Generate shield-config.cmake

File: recipes/shield/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,32 @@
+from conans import ConanFile, tools, CMake
+import os
+
```
Comment (uilianries) [inline recipes/shield/all/conanfile.py:3]:
```suggestion

required_conan_version = ">=1.33.0"

```

Comment (lkotsonis) [reply recipes/shield/all/conanfile.py:3]: Done

File: recipes/shield/all/conanfile.py (hunk 2 above)
Comment (lkotsonis) [reply recipes/shield/all/conanfile.py]: done

File: recipes/shield/all/conanfile.py (hunk 3 above)
Comment (lkotsonis) [reply recipes/shield/all/conanfile.py]: done

File: recipes/shield/all/test_package/CMakeLists.txt (hunk 5 above)
Comment (lkotsonis) [reply recipes/shield/all/test_package/CMakeLists.txt]: done

**Review by uilianries — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

File: recipes/shield/all/conanfile.py [hunk 8]
```diff
@@ -0,0 +1,32 @@
+from conans import ConanFile, tools, CMake
```
[the remark this answers is not in the document]
Comment (lkotsonis) [reply recipes/shield/all/conanfile.py]: done!

**Review by uilianries — APPROVED** [review]
LGTM

File: recipes/shield/all/conanfile.py [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+    homepage = "https://github.com/holoplot/shield"
+    license = "MIT"
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def source(self):
+     tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
```
Comment (uilianries) [inline recipes/shield/all/conanfile.py:21]:
```suggestion
        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
```
Bad indentation

Comment (lkotsonis) [reply recipes/shield/all/conanfile.py]: fixed

**Review by uilianries — APPROVED** [review]
LGTM

**Review by sunbubble — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Comment by lkotsonis** [comment]
Seems like the CI is stuck, is there anything that I can do (besides force pushing) to retrigger?

**Review by jgsogo — APPROVED** [review]
I've just triggered the CI again

File: recipes/shield/all/conanfile.py [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+    def package(self):
+        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
+        self.copy(pattern="*", dst="include", src=os.path.join(self._source_subfolder, "include"))
+
+    def package_id(self):
+        self.info.header_only()
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "shield"
+        self.cpp_info.names["cmake_find_package_multi"] = "shield"
```
Comment (SpaceIm) [inline recipes/shield/all/conanfile.py:32]: There is no CMake config file upstream, so these values shouldn't be enforced.

Comment (lkotsonis) [reply recipes/shield/all/conanfile.py:32]: So, removing them would still work, right? I can still make a PR. I think I copied this from one of the recommended recipes to take as an example.

Comment (SpaceIm) [reply recipes/shield/all/conanfile.py:32]: yes

**Pull request merged**